test/src/unit-cppapi-consolidation-with-timestamps.cc

```diff
@@ -74,15 +74,17 @@ struct ConsolidationWithTimestampsFx {
       std::vector<uint64_t>& dim2,
       std::string& stats,
       tiledb_layout_t layout,
-      uint64_t timestamp);
+      uint64_t timestamp,
```

Comment:
I think all your test cases read fragments consolidated with timestamps correct? We should add tests that request the timestamp attribute for normal fragments.

I'm also asking because I don't see code that processes fragments without timestamps. Look at SparseGlobalOrderReader::copy_timestamps_tiles, we most likely need to implement a similar function in sparse_unordered_with_dups_reader.cc.